flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/ExecutionCheckpointingOptions.java

```diff
@@ -280,4 +280,13 @@ public class ExecutionCheckpointingOptions {
                     .booleanType()
                     .defaultValue(false)
                     .withDescription("Flag to enable approximate local recovery.");
+
+    public static final ConfigOption<Integer> CHANNEL_STATE_NUMBER_OF_SUBTASKS_SHARE_FILE =
+            ConfigOptions.key("execution.checkpointing.channel-state.number-of-subtasks-share-file")
+                    .intType()
+                    .defaultValue(5)
+                    .withDescription(
+                            "Defines the maximum number of subtasks that share the same channel state file. "
+                                    + "It can reduce the number of small files when enable unaligned checkpoint. "
+                                    + "Each subtask will create a new channel state file when this is configured to 1.");
```

Comment:
> number-of-subtasks-share-file

This sounds a bit strange in english. Maybe let's rename it to:

> execution.checkpointing.channel-state.subtasks-per-file

? and renaming the config option and getters as well? @dawidwys maybe you have some better idea?

Reply:
1. Does it apply to unaligned checkpoints only? If so, could we put it under `unaligned` subgroup? `execution.checkpointing.unaligned.(...)`
2. Do we plan to have more options under `channel-state`? If not, I would not introduce this subgroup.
3. I like @pnowojski idea more, but I'd add `max` prefix: `max-subtasks-per-file`

Depending on 2. I'd go with either `channel-state.max-subtasks-per-file` or ` max-subtasks-per-channel-state-file`

Reply:
It's only for unaligned checkpoints. There is a possibility we will have more options, but not very likely. So maybe `execution.checkpointing.unaligned.max-subtasks-per-channel-state-file`?

Reply:
Makes sense to me

Reply:
Sounds good to me.

Thanks for the suggestion, I will update it asap.